### PR TITLE
draft: [CAI-204] Chatbot/presidio in docker for local development

### PR DESCRIPTION
**Pull request by batdevis**
#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/chatbot/docker/app.local.Dockerfile [hunk 1]
```diff
@@ -14,4 +18,10 @@ RUN poetry install
 
 COPY . .
 
-CMD ["fastapi", "dev", "src/app/main.py", "--port", "8080"]
+RUN python -m spacy download it_core_news_md
+RUN python -m spacy download en_core_web_md
```
Comment (mdciri) [inline apps/chatbot/docker/app.local.Dockerfile:22]: here are missing to download: `es_core_news_md`, `de_core_news_md`, `fr_core_news_md`,  `el_core_news_md`, `hr_core_news_md`, and `sl_core_news_md`.

File: apps/chatbot/pyproject.toml [hunk 2]
```diff
@@ -37,7 +37,10 @@ llama-index-llms-gemini = "^0.3.4"
 google-generativeai = "^0.5.2"
 llama-index-embeddings-gemini = "^0.2.0"
 llama-index-llms-bedrock-converse = "^0.3.0"
+chromedriver-py = "^129.0.6668.91"
 llama-index-postprocessor-presidio = "^0.2.0"
+presidio-anonymizer = "^2.2.355"
+presidio-analyzer = "^2.2.355"
```
Comment (mdciri) [inline apps/chatbot/pyproject.toml:43]: remove `presidio-anonymizer` and `presidio-analyzer` and install `llama-index-postprocessor-presidio` as in the `pyproject.toml` in main. Check this out.

In the end, why `chromedriver-py`?

File: apps/chatbot/src/app/main.py [hunk 3]
```diff
@@ -18,30 +18,30 @@
...
-if (os.getenv('environment', 'dev') == 'local'):
-  profile_name='dummy'
-  endpoint_url='http://localhost:8000'
-  region_name = AWS_DEFAULT_REGION
-
 boto3_session = boto3.session.Session(
-  profile_name = locals().get('profile_name', None),
-  region_name=locals().get('region_name', None)
+  region_name=AWS_DEFAULT_REGION
 )
```
Comment (mdciri) [inline apps/chatbot/src/app/main.py:32]: in boto3 session are missing the `aws_access_key_id` and `aws_secret_access_key`

File: apps/chatbot/src/app/main.py [hunk 4]
```diff
@@ -18,30 +18,30 @@
...
 
+chatbot = Chatbot(params, prompts)
+
 
 class Query(BaseModel):
   question: str
   queriedAt: str | None = None
 
-if (os.getenv('environment', 'dev') == 'local'):
-  profile_name='dummy'
```
Comment (mdciri) [inline apps/chatbot/src/app/main.py:31]: I saw this `dummy` in many files. isn't better to but this `profile_name` in `config/params.yaml` or in a enviroment variable?

File: apps/chatbot/src/app/main.py [hunk 5]
```diff
@@ -18,30 +18,30 @@
...
-  'dynamodb',
-  endpoint_url=locals().get('endpoint_url', None),
-  region_name=locals().get('region_name', None),
-)
+if (os.getenv('environment', 'dev') == 'local'):
+  dynamodb = boto3_session.resource(    
+    'dynamodb',
+    endpoint_url=os.getenv('CHB_DYNAMODB_URL', 'http://localhost:8000'),
+    region_name=AWS_DEFAULT_REGION
+  )
```
Comment (mdciri) [inline apps/chatbot/src/app/main.py:39]: `aws_access_key_id` and `aws_secret_access_key` are missing

File: apps/chatbot/src/app/main.py [hunk 6]
```diff
@@ -18,30 +18,30 @@
...
+else:
+  dynamodb = boto3_session.resource(    
+    'dynamodb',
+    region_name=AWS_DEFAULT_REGION
+  )
```
Comment (mdciri) [inline apps/chatbot/src/app/main.py:45]: `aws_access_key_id` and `aws_secret_access_key` are missing

File: apps/chatbot/src/modules/utils.py [hunk 7]
```diff
@@ -19,6 +19,7 @@ def get_ssm_parameter(name: str, default: str | None = None) -> str | None:
     :param default: The default value to return if the parameter is not found.
     :return: The value of the requested parameter.
     """
+
     ssm = boto3.client(
         "ssm",
         aws_access_key_id=AWS_ACCESS_KEY_ID,
```
Comment (mdciri) [inline apps/chatbot/src/modules/utils.py:25]: `aws_access_key_id` and `aws_secret_access_key` are missing, but this is already handled in `main`

File: apps/chatbot/src/modules/vector_database.py [hunk 8]
```diff
@@ -166,17 +166,27 @@ def create_documentation(
 
     for file in tqdm.tqdm(html_files, total=len(html_files), desc="Extracting HTML"):
 
-        if file in dynamic_htmls:
+# FIX: resolve webdriver.Chrome "self.assert_process_still_running" error in docker
+#        if file in dynamic_htmls:
+        if 6 == 9:
```
Comment (mdciri) [inline apps/chatbot/src/modules/vector_database.py:171]: fix this

**Review by mdciri — CHANGES_REQUESTED** [review]

**Comment by batdevis** [comment]
now it works with Presidio

**Pull request closed without merging**